handlers/handlers.go

```diff
@@ -202,3 +214,110 @@ func ListPageRender(rend RenderClient, cli CodeListClient) http.HandlerFunc {
 		return
 	}
 }
+
+//AreaPageRender ...
+func AreaPageRender(rend RenderClient, cli CodeListClient, dcli DatasetClient) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		vars := mux.Vars(req)
+		codeListID := vars["codeListID"]
+		codeID := vars["codeID"]
+		logData := log.Data{
+			codeListID: codeListID,
+			codeID:     codeID,
+		}
+		var page area.Page
+
+		codeListEditions, err := cli.GetCodeListEditions(codeListID)
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "error getting editions for a code-list"), logData)
+			setStatusCode(req, w, err)
+			return
+		}
+
+		var parentName string
+
+		if codeListEditions.Count > 0 {
+			edition := codeListEditions.Items[0]
+			parentName = edition.Label
+
+			log.InfoCtx(ctx, "getting data about code", log.Data{"edition": edition})
+			codeData, err := cli.GetCodeByID(codeListID, edition.Edition, codeID)
+			if err != nil {
+				log.ErrorCtx(ctx, errors.WithMessage(err, "error getting code data"), logData)
+				setStatusCode(req, w, err)
+				return
+			}
+			page.Metadata.Title = codeData.Label
+
+			datasetsResp, err := cli.GetDatasetsByCode(codeListID, edition.Edition, codeID)
+			if err != nil {
+				log.ErrorCtx(ctx, errors.WithMessage(err, "error getting datasets related to code"), logData)
+				setStatusCode(req, w, err)
+				return
+			}
+
+			if datasetsResp.Count > 0 {
+				var datasets []area.Dataset
+				for _, datasetResp := range datasetsResp.Datasets {
```

Comment:
I could be wrong, but would this loop benefit from being a go routine instead? It means we could send all of the requests off at one time, so should improve performance.